### PR TITLE
Enable metrics publishing for mt-gateway

### PR DESCRIPTION
**Pull request by fitzoh**
As requested [here](https://github.com/grafana/metrictank/pull/1644#discussion_r372038457).

Note that there are a couple places where prometheus instrumentation is used that have not been addressed:
https://github.com/grafana/metrictank/blob/d37ddf3e338858f26da7a4bc469b383a2ab7cc07/cmd/mt-gateway/ingest/metrics.go#L29
https://github.com/grafana/metrictank/blob/d37ddf3e338858f26da7a4bc469b383a2ab7cc07/publish/publish.go#L13

### CODE REVIEW

File: cmd/mt-gateway/main.go [hunk 1]
```diff
@@ -23,6 +26,14 @@ var (
 	importerURL   = flag.String("importer-url", "", "mt-whisper-importer-writer address")
 	defaultOrgId  = flag.Int("default-org-id", -1, "default org ID to send to downstream services if none is provided")
 	brokers       = flag.String("kafka-tcp-addr", "localhost:9092", "kafka tcp address(es) for metrics, in csv host[:port] format")
+
+	// stats
+	statsEnabled    = flag.Bool("stats-enabled", false, "enable sending graphite messages for instrumentation")
+	statsPrefix     = flag.String("stats-prefix", "tsdb-gw.stats.default.$hostname", "stats prefix (will add trailing dot automatically if needed)")
```
Comment (replay) [inline cmd/mt-gateway/main.go]: the default value here does not match the one in the config file. I think it should also be `mt-gateway.stats.default.$hostname`

**Review by replay — CHANGES_REQUESTED** [review]
LGTM, just one very minor issue to fix

File: cmd/mt-gateway/main.go (hunk 1 above)
Comment (fitzoh) [reply cmd/mt-gateway/main.go]: oops, e5ee174d386d81815f60907794fea32d068562ef

**Review by replay — APPROVED** [review]
:shipit:

**Comment by Dieterbe** [comment]
before we merge. can you confirm that you can take a tsdb-gw dashboard, `s/tsdb-gw/mt-gateway/` and it works ?

**Comment by fitzoh** [comment]
Screen Shot 2020-01-29 at 10 30 40 AM

First couple look good, http stuff isn't instrumented

**Comment by Dieterbe** [comment]
>  http stuff isn't instrumented

can we ?

**Comment by fitzoh** [comment]
>can we ?

Copied over in c803c82a50538b85e9aa3376c2c58022bcb925d2... Not sure if we want to leave them as is or switch to tag based metrics.

**Comment by Dieterbe** [comment]
how is this copied over? the code looks quite different, making it hard to compare.. can you just make a literal copy, and maybe make any needed changes in a separate commit?

also, that screenie looks good. can you confirm these stats similarly?

**Review by Dieterbe — CHANGES_REQUESTED** [review]
how is this copied over? the code looks quite different, making it hard to compare.. can you just make a literal copy, and maybe make any needed changes in a separate commit?

also, that screenie looks good. can you confirm these stats similarly?

**Comment by fitzoh** [comment]
>how is this copied over? the code looks quite different, making it hard to compare.. can you just make a literal copy, and maybe make any needed changes in a separate commit?

Initial import in https://github.com/grafana/metrictank/pull/1645/commits/1b16b0d992cc13832351b4f173a8b50635a30367, changes in subsequent commits

**Comment by fitzoh** [comment]
>also, that screenie looks good. can you confirm these stats similarly?

After some tweaks to get things up and running...

Screen Shot 2020-02-10 at 11 09 01 AM

The response codes were worrying me a little bit, but I was able to confirm that there's data there through the explore view. (The query the dashboard was running was `groupByNode(perSecond(mt-gateway.stats.$env.$host.api.request.$path.status.*.counter32), 8, 'sum')`)

Screen Shot 2020-02-10 at 11 09 59 AM

**Review by Dieterbe — APPROVED** [review]

**Pull request merged**